### PR TITLE
Move tests into the `tests` directory & run tests on GH Actions

### PR DESCRIPTION
**Pull request by sampoder**
The successor to #121... rebasing is doing my head in but full steam ahead! This PR works towards resolving #82 & #60.

### CODE REVIEW

**Review by conradev — CHANGES_REQUESTED** [review]
We need to fix two things here:

- The test command should match the build command more. We should test all targets and features. We should also make explicit which targets we are testing in the matrix configuratio (i.e. the host target)

- The test command seemingly recompiles the crate, I'd be curious why it can't re-use the previous compile (might be for the above reason)

**Comment by sampoder** [comment]
Let me look into recompilation, that'll speed things up a lot when removed!

File: .github/workflows/build-rust.yml [hunk 1]
```diff
@@ -60,3 +60,6 @@ jobs:
     - name: Build
       shell: bash
       run: cargo build --verbose --workspace --all-features --target ${{ join(matrix.targets, ' --target ') }}
+    - name: Post-Build Tests
+      shell: bash
+      run: cargo test
```
Comment (sampoder) [inline .github/workflows/build-rust.yml:65]:
```suggestion
      run: cargo test --verbose --workspace --all-features --target ${{ join(matrix.targets, ' --target ') }}
```

@conradev is this what you're looking for?

**Comment by sampoder** [comment]
> The test command seemingly recompiles the crate, I'd be curious why it can't re-use the previous compile (might be for the above reason)

Based on https://github.com/sagiegurari/cargo-make/issues/251, it seems like removing `cargo build [...]` and just doing `cargo test [...]` would achieve what we'd want.

**Review by conradev — APPROVED** [review]

**Pull request merged**